strangerstudios/paid-memberships-pro
# adding CardinalCommerce fields to gateway settings

### All Submissions:






### Changes proposed in this Pull Request:



Adding the following CardinalCommerce fields to Payment Gateway & SSL Settings in preparation for SCA updates:

API Key
API Identifier
OrgUnit ID
SongbirdUrl
Merchant ID
Processor ID

Added separate CardinalCommerce settings for each core payment gateway that uses CardinalCommerce. Based on the documentation I've seen, this includes all Paypal flavors, Cybersource, possibly Authorize.net (seeing conflicting reports), and Braintree. Please comment as needed if some of those should be pruned or added.

NOTE: Paypal Standard, Paypal Express, and Paypal Website Payments Pro share the same settings. See here: https://github.com/strangerstudios/paid-memberships-pro/blob/dev/classes/gateways/class.pmprogateway_paypal.php#L33-L38 If we need them to have separate settings, will need to refactor how those three have their payment option fields added to page.

### How to test the changes in this Pull Request:

1. Go to Memberships > Settings > Payment Gateway & SSL
2. Choose a gateway from the above list 
3. Enter CardinalCommerce details and save settings
4. Check pmpro_options table for the new values (saved under `pmpro_[GATEWAY NAME]_cardinal_[OPTION NAME]`)
5. Reload Payment Settings and check that settings match those from steps 3 + 4

### Other information:


WP 5.2.2, PMPro 2.1, PHP 7.2.15



### Changelog entry

-Added CardinalCommerce settings to Payment Gateway & SSL settings for core gateways that integrate with CardinalCommerce for Secure Customer Authentication.
